src/pages/signin/ResendValidationForm.js

```diff
@@ -123,7 +114,7 @@ class ResendValidationForm extends React.Component {
                         success
                         text={this.props.translate('resendValidationForm.resendLink')}
                         isLoading={this.props.account.loading}
-                        onPress={this.validateAndSubmitForm}
+                        onPress={this.onSubmit}
```

Comment:
is there a reason we can't get rid of this `onSubmit` method and inline the logic of that method like?
```suggestion
                        onPress={() => (this.props.account.validated ? Session.resetPassword() : Session.resendValidationLink())}
```

Reply:
I figured since we had a function doing the submit before that I'd keep the same pattern, but it honestly is cleaner to just do it inline. I'll update!